Source/UnrealExamples/ReadyPlayerMeWebBrowser.cpp

```diff
@@ -15,36 +20,84 @@ static const TCHAR* GenderFemaleParam = TEXT("gender=female");
 
 static const TMap<ELanguage, FString> LANGUAGE_TO_STRING =
 {
-	{ ELanguage::En, "en" },
-	{ ELanguage::EnIe, "en-IE" },
-	{ ELanguage::De, "de" },
-	{ ELanguage::Fr, "fr" },
-	{ ELanguage::Es, "es" },
-	{ ELanguage::EsMx, "es-MX" },
-	{ ELanguage::Pt, "pt" },
-	{ ELanguage::PtBr, "pt-BR" },
-	{ ELanguage::It, "it" },
-	{ ELanguage::Tr, "tr" },
-	{ ELanguage::Jp, "jp" },
-	{ ELanguage::Kr, "kr" },
-	{ ELanguage::Ch, "ch" }
+	{ELanguage::En, "en"},
+	{ELanguage::EnIe, "en-IE"},
+	{ELanguage::De, "de"},
+	{ELanguage::Fr, "fr"},
+	{ELanguage::Es, "es"},
+	{ELanguage::EsMx, "es-MX"},
+	{ELanguage::Pt, "pt"},
+	{ELanguage::PtBr, "pt-BR"},
+	{ELanguage::It, "it"},
+	{ELanguage::Tr, "tr"},
+	{ELanguage::Jp, "jp"},
+	{ELanguage::Kr, "kr"},
+	{ELanguage::Ch, "ch"}
 };
 
-void UReadyPlayerMeWebBrowser::SetupBrowser(const FReadyPlayerWebBrowserResponse& Response)
+void UReadyPlayerMeWebBrowser::SetupBrowser()
 {
 	BindBrowserToObject();
-	WebLinkObject->SetAvatarUrlCallback(Response);
-	ExecuteJavascript(JSAddAvatarGeneratedListener);
+	const FString Path = FPaths::ProjectContentDir() / JavascriptPath;
+
+
+	FString rpmSetupJavascript;
+	if (FPaths::FileExists(Path))
+	{
+		FFileHelper::LoadFileToString(rpmSetupJavascript, *Path);
+	}
+	else
+	{
+		UE_LOG(LogTemp, Warning, TEXT("Script file not found: %s"), *Path);
+		return;
+	}
+
+	ExecuteJavascript(rpmSetupJavascript);
 }
 
 void UReadyPlayerMeWebBrowser::BindBrowserToObject()
 {
-	if (!WebLinkObject) {
+	if (!WebLinkObject)
+	{
 		WebLinkObject = NewObject<UWebLink>(this, *LinkObjectName);
 	}
+	WebLinkObject->SetWebBrowser(*this);
```

Comment:
Now that the UWebLink doesn't have much logic maybe it's better to bind the event to the WebBrowser instead?
```
WebBrowserWidget->BindUObject(LinkObjectName, this);
```
then move the `void EventReceived(const FString JsonResponse);` from the UWebLink to UReadyPlayerMeWebBrowser

Reply:
Wait I think you can not WebBrowserWidget doesn't inherit from the UObject.

Reply:
hmm good point, didn't think of that, will try it out and change if it works.